system/lib/libc/musl/src/thread/pthread_detach.c

```diff
@@ -3,10 +3,11 @@
 
 static int __pthread_detach(pthread_t t)
 {
-	// XXX EMSCRIPTEN: Add check for invalid (already joined) thread.  Again
-	// for the benefit of the conformance tests.
-	if (t->self != t)
-		return ESRCH;
+#ifdef __EMSCRIPTEN__
+	// Attempt to detach a thread which does not point to a valid thread, or
+	// does not exist anymore.
+	if (!t || t->self != t) return ESRCH;
```

Comment:
Is the test for NULL here needed somewhere?   Most other pthread function don't have this kind of check IIRC.

If it is needed perhaps this can be its own change?

Reply:
The reason for this change was that the previous JS code did check for this, see e.g:
https://github.com/emscripten-core/emscripten/blob/85d58fff45cf47de8df3df2925c6c64e4281b6f7/src/library_pthread.js#L906-L909

I removed this NULL check with commit https://github.com/emscripten-core/emscripten/pull/15604/commits/80a3e1108c81797aa42fca835adcf0d2db9734a0 as it does indeed looks redundant.